File: backend/app/schemas/card.py
import json
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, computed_field


class CardSummary(BaseModel):
    id: int
    name: str
    set_name: str
    number: str
    rarity: Optional[str] = None
    image_small: Optional[str] = None
    lowest_price: Optional[float] = None
    lowest_price_marketplace: Optional[str] = None

    model_config = {"from_attributes": True}


class CardDetail(BaseModel):
    id: int
    pokemon_tcg_id: str
    name: str
    set_name: str
    set_code: str
    number: str
    rarity: Optional[str] = None
    supertype: str
    image_small: Optional[str] = None
    image_large: Optional[str] = None
    artist: Optional[str] = None
    hp: Optional[str] = None
    types_json: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

    @computed_field
    @property
    def types(self) -> list[str]:
        if self.types_json:
            try:
                return json.loads(self.types_json)
            except (json.JSONDecodeError, TypeError):
                pass
        return []


class PaginatedCardResponse(BaseModel):
    items: list[CardSummary]
    total: int
    page: int
    page_size: int
    total_pages: int

File: frontend/src/types/card.ts
export interface CardSummary {
  id: number
  name: string
  set_name: string
  number: string
  rarity: string | null
  image_small: string | null
  lowest_price: number | null
  lowest_price_marketplace: string | null
}

export interface CardDetail {
  id: number
  pokemon_tcg_id: string
  name: string
  set_name: string
  set_code: string
  number: string
  rarity: string | null
  supertype: string
  image_small: string | null
  image_large: string | null
  artist: string | null
  hp: string | null
  types: string[]
  created_at: string
  updated_at: string
}

export interface CardSuggestion {
  id: number
  name: string
  set_name: string
  image_small: string | null
}

export interface PaginatedCards {
  items: CardSummary[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface SetInfo {
  id: string
  name: string
  series: string
  total: number
  releaseDate: string
  images: {
    symbol: string
    logo: string
  }
}
